apps/src/gamelab/AnimationPicker/animationPickerModule.js

```diff
@@ -110,27 +110,34 @@ export function handleUploadComplete(result) {
     const { goal, uploadFilename } = getState().animationPicker;
     const key = result.filename.replace(/\.png$/i, '');
     const sourceUrl = animationsApi.basePath(key + '.png');
-
-    loadImageMetadata(sourceUrl, metadata => {
-      const animation = _.assign({}, metadata, {
-        name: uploadFilename,
-        sourceUrl: sourceUrl,
-        size: result.size,
-        version: result.versionId
-      });
-
-      if (goal === Goal.NEW_ANIMATION) {
-        dispatch(addAnimation(key, animation));
-      } else if (goal === Goal.NEW_FRAME) {
-        dispatch(appendCustomFrames(animation));
-      }
-      dispatch(hide());
-    }, () => {
+    const onImageMetadataLoaded = buildOnImageMetadataLoaded(uploadFilename, goal, key, result, dispatch);
+    // TODO (bbuchanan): This sequencing feels backwards.  Eventually, we
+    // ought to preview and get dimensions from the local filesystem, async
+    // with the upload itself, but that will mean refactoring away from the
+    // jQuery uploader.
```

Comment:
I tore out this comment in https://github.com/code-dot-org/code-dot-org/pull/20713 - let's not reintroduce it.